empress/newickTreeReader.py

```diff
@@ -0,0 +1,215 @@
+# ReconInput class
+from io import StringIO
+
+# BioPython libraries
+from Bio import Phylo
+
+class ReconInput(object):
+    """
+    Storage class for the newick data (trees, tip mapping, and optional distance parameters)
+    Distances encode branch lengths in the newick file, if given
+    """
+
+    def __init__(self, host_tree = None, host_distances = None, parasite_tree = None, 
+                parasite_distances = None, phi = None):
+        self.host_tree = host_tree
+        self.host_distances = host_distances
+        self.parasite_tree = parasite_tree
+        self.parasite_distances = parasite_distances
+        self.phi = phi
+
+    def complete(self):
+        return not self.host_tree is None and not self.parasite_tree is None and not self.phi is None
+
+    def read_host(self, file_name : str):
+        """
+        Takes a host filename as input and sets self.host_dict and self.host_distances
+        :param file_name <str>    - filename of host file to parse
+        :return None
+        """
+        self.host_tree, self.host_distances = ReconInput.read_newick_tree(file_name, "host")
+  
+    def read_parasite(self, file_name : str):
+        """
+        Takes a parasite filename as input and sets self.parasite_dict and self_host_distances
+        :param file_name <str>   - filename of parasite file to parse
+        :return None
+        """
+        self.parasite_tree, self.parasite_distances = ReconInput.read_newick_tree(file_name, "parasite")
+
+    def read_mapping(self, file_name : str):
+        """
+        Takes a mapping filename as input and reutrns the mapping dictionary between a host and parasite
+        :param filename <str>   - filename of the map file to parse
+        :return None
+        """
+        if isinstance(file_name, str):
+            map_file = open(file_name, 'r')
+        else:
+            return None
+
+        map_string = map_file.read()
+        map_list = map_string.split()
+        self.phi = ReconInput.parse_phi(map_list)
+        map_file.close()
+
+    @staticmethod
+    def imply_mapping(host_tree : tree, parasite_tree : tree):
+        """
```

Comment:
It seems like this function can produce an invalid mapping if some of names follow the convention while some of the names doesn't follow the convention. If the inferred mapping is invalid, I think we should return `None` instead of the invalid mapping.